src/jit/lowerxarch.cpp

```diff
@@ -2838,10 +2835,10 @@ void Lowering::TreeNodeInfoInitSIMD(GenTree* tree)
             // registers. Note that if this is a TYP_SIMD16 or smaller on AVX, then we
             // don't need a tmpReg.
             //
-            // 32-byte integer vector on AVX:
+            // 32-byte integer vector on SSE4/AVX:
             // will take advantage of phaddd, which operates only on 128-bit xmm reg.
-            // This would need 2 internal registers since targetReg is an int type
-            // register.
+            // This would need 1 (in case of SSE4) or 2 (in case of AVX) internal
```

Comment:
Since you are here, you should change this to "This will need", as it is not conditional on anything.